src/components/PermissionsDialog/PermissionsDialog.react.js

```diff
@@ -224,22 +224,26 @@ function renderPointerCheckboxes(rowId, publicPerms, pointerPerms, advanced, onC
 }
 
 export default class PermissionsDialog extends React.Component {
-  constructor(props) {
+  constructor({
+    enablePointerPermissions = false,
+    permissions,
+    advanced,
+  }) {
     super();
 
     let uniqueKeys = ['*'];
     let perms = {};
-    for (let k in props.permissions) {
+    for (let k in permissions) {
```

Comment:
I'm actually not a huge fan of using reduce to build up lists/objects. It feels like a lot of boilerplate compared to a for loop. Also this is just a change to use destructuring, I don't really want to go changing other stuff in this function.